addons/mrp/models/stock_move.py

```diff
@@ -225,6 +229,12 @@ def _onchange_product_uom_qty(self):
             mo = self.raw_material_production_id
             self._update_quantity_done(mo)
 
+    @api.onchange('quantity_done')
+    def _onchange_quantity_done(self):
```

Comment:
Do compute instead of onchange when possible since we would like to have something similar in a forw view and create call in the future

Reply:
The issue is in `self.should_consume_qty != self.quantity_done` both values are computed. When update `qty_to_produce`, those two values will be updated in a weird way and make this condition `True`. Onchange avoided those updates.

Reply:
The dependencies tree should be built correctly. Could you check a bit deeper